benchmark/checks/check_dialogues_memory_usage.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""Memory usage of dialogues across the time."""
+import os
+import sys
+import time
+import uuid
+from typing import cast
+
+import click
+
+from aea.common import Address
+from aea.protocols.base import Message
+from aea.protocols.dialogue.base import Dialogue
+from benchmark.checks.utils import get_mem_usage_in_mb  # noqa: I100
+from benchmark.checks.utils import multi_run, print_results
+
+from packages.fetchai.protocols.http.dialogues import HttpDialogue, HttpDialogues
+from packages.fetchai.protocols.http.message import HttpMessage
+
+
+ROOT_PATH = os.path.join(os.path.abspath(__file__), "..", "..")
+sys.path.append(ROOT_PATH)
+
+
+class DialogueHandler:
+    """Generate messages and process with dialogues."""
+
+    def __init__(self):
+        """Set dialogues."""
+        # pylint: disable=unused-argument
+
+        def role(m: Message, addr: Address) -> Dialogue.Role:
+            return HttpDialogue.Role.CLIENT
+
+        self.addr = self.random_string
+        self.dialogues = HttpDialogues(self.addr, role_from_first_message=role)
+
+    @property
+    def random_string(self) -> str:
+        """Get random string on every access."""
+        return uuid.uuid4().hex
+
+    def process_message(self) -> None:
+        """Process a message with dialogues."""
+        message = self.create()
+        dialogue = self.update(message)
+        self.reply(dialogue, message)
```

Comment:
This is wrong. The dialogues model has a `create` method. use this, see the skills for examples!

Reply:
e.g. generic buyer skill, behaviours.py line 81

Reply:
every message generated is sent from a new address.
so the message has to be generated manually,  existing dialogues can not be used, cause the same dialogue can not be used for creating and reply (checked!)

Reply:
Yes you'd need to create two dialogues. That's better though as it keeps it consistent. The message should be created outside the handler (e.g. in a behaviour) and just then processed by the handler. Your custom logic here does not reflect what we have in the framework.

Reply:
realistic case implemented in `check_multiagent_http_dialogues.py`, a limited amount of agents are interacting, there is one agent interacting with like an unlimited amount of agent, so every message generated from the new address